tests/Baseline/spicy.optimization.default-parser-functions/log

```diff
@@ -307,7 +300,7 @@
 [debug/optimizer] [default-parser-functions.spicy:10:1-21:2] declaration::Constant "const bool __feat%foo@@P2%uses_random_access = False;" -> disabled feature 'uses_random_access' of type 'foo::P2' since it is not used
 [debug/optimizer] [default-parser-functions.spicy:10:1-21:2] declaration::Constant "const bool __feat%foo@@P2%uses_stream = False;" -> disabled feature 'uses_stream' of type 'foo::P2' since it is not used
 [debug/optimizer] [default-parser-functions.spicy:10:1-21:2] declaration::Constant "const bool __feat%foo@@P2%uses_sync_advance = False;" -> disabled feature 'uses_sync_advance' of type 'foo::P2' since it is not used
-[debug/optimizer] [default-parser-functions.spicy:12:1-12:18] declaration::Type "type P0 = struct { optional<hilti::RecoverableFailure> __error &always-emit &internal; method tuple<const view<stream>, int<64>, const iterator<stream>, optional<hilti::RecoverableFailure>> __parse_foo__P0_stage2(); } &on-heap;" -> null (removing unused type)
+[debug/optimizer] [default-parser-functions.spicy:12:1-12:18] declaration::Type "type P0 = struct { optional<hilti::RecoverableFailure> __error &always-emit &internal; method tuple<const view<stream>, int<64>, const iterator<stream>, optional<hilti::RecoverableFailure>> __parse_foo__P0_stage2(inout value_ref<stream> __data, iterator<stream> __begin, copy view<stream> __cur, copy bool __trim, copy int<64> __lah, copy iterator<stream> __lahe, copy optional<hilti::RecoverableFailure> __error); } &on-heap;" -> null (removing unused type)
```

Comment:
For my understanding, where is this difference for the stage2 parameters are coming from? Is this just due to a change in the order of the optimization passes?  Seeing many of among the baseline diffs, and also some related changes where parameters used to be removed but are no longer logged as such.

Reply:
Yes I believe so, I only see this in the logging, but the generated code stays the same. This is due to running the existing `FunctionParamVisitor` now after all other existing passes so the previous phase has already removed the function before we can try to prune its parameters. If I move `FunctionParamVisitor` to the initial phase I see the previous log output.